mam/v2/mss.c

```diff
@@ -468,27 +460,31 @@ bool_t mss_verify(isponge *ms, isponge *ws, trits_t H, trits_t sig,
 #if defined(MAM2_MSS_DEBUG)
   trits_copy(trits_take(sig, MAM2_MSS_MT_HASH_SIZE), apk);
 #else
-  MAM2_ASSERT(trits_size(pk) == MAM2_WOTS_PK_SIZE);
+  MAM2_ASSERT(pk->num_trits == MAM2_WOTS_PK_SIZE);
   TRIT_ARRAY_MAKE_FROM_RAW(hash_array, MAM2_WOTS_HASH_SIZE, H.p + H.d);
   TRIT_ARRAY_MAKE_FROM_RAW(sk_sig_array, MAM2_WOTS_SIG_SIZE, sig.p + sig.d);
-  TRIT_ARRAY_DECLARE(pk_array, MAM2_WOTS_PK_SIZE);
-  wots_recover(ws, &hash_array, &sk_sig_array, &pk_array);
-  flex_trits_to_trits(apk.p + apk.d, MAM2_WOTS_PK_SIZE, pk_array.trits,
-                      MAM2_WOTS_PK_SIZE, MAM2_WOTS_PK_SIZE);
+  TRIT_ARRAY_DECLARE(apk, MAM2_MSS_MT_HASH_SIZE);
+  wots_recover(ws, &hash_array, &sk_sig_array, &apk);
+
 #endif
   dbg_printf("\nwpR\t");
-  trits_dbg_print(apk);
+  // trits_dbg_print(apk);
   sig = trits_drop(sig, MAM2_WOTS_SIG_SIZE);
 
-  mss_fold_apath(ms, skn, sig, apk);
+  TRIT_ARRAY_MAKE_FROM_RAW(ap_array, sig.n - sig.d, sig.p + sig.d);
+
+  mss_fold_apath(ms, skn, &ap_array, &apk);
```

Comment:
#556 Introduces flex auth path, maybe be easier to merge it before so no need for conversions